src/Models/BaseElement.php

```diff
@@ -895,7 +895,9 @@ public function getAreaRelationName()
 
         if ($page) {
             $has_one = $page->config()->get('has_one');
-            $area = $this->Parent();
+            $class = DataObject::getSchema()->hasOneComponent($this, 'Parent');
+            $area = ($this->ParentID) ? DataObject::get_by_id($class, $this->ParentID) : null;
+            $area = $area ?: ElementalArea::singleton();
```

Comment:
What's the idea behind getting a singleton here? This feels like a bit of a hack tbh. The get_by_id part is sensible but I don't understand why you're getting a singleton if that fails.

Assuming it is sensible, we should use the null coallescing asignment operator instead of `?:` since this shouldn't be falsy unless it's explicitly null.
```suggestion
            $area ??= ElementalArea::singleton();
```

Reply:
> What's the idea behind getting a singleton here? This feels like a bit of a hack tbh. The get_by_id part is sensible but I don't understand why you're getting a singleton if that fails.

Just matching the `$this->Parent()` call - `DataObject::getComponent()` will return an empty record if the record doesn‘t exist. I suppose we could remove that and wrap the code that follows in an `if ($area)` check instead?

Reply:
I’ve replaced the singleton logic with an `if ($area)` statement

Reply:
Thanks! I think that's more straightforward and more consistent with similar logic elsewhere in core and supported modules.